var/spack/repos/builtin/packages/infer/package.py

```diff
@@ -0,0 +1,118 @@
+    # version('0.13.0', 'd6e18dbcfa846feb1547bff4dbdd96f0')
+
+    depends_on('autoconf', type=('build'))
+    depends_on('automake', type=('build'))
+    depends_on('cmake', type=('build'))
+    depends_on('libtool', type=('build'))
+    depends_on('ocaml', type=('build'))
+    depends_on('opam', type=('build'))
+    depends_on('pkg-config', type=('build'))
+
+    parallel = False
+
+    def setup_environment(self, spack_env, run_env):
+        spack_env.set('OPAMYES', '1')
+        spack_env.set('INFER_CONFIGURE_OPTS',
+                      ('--prefix={0}'.format(self.prefix) +
+                       ' --disable-ocaml-binannot' +
+                       ' --without-fcp-clang'))
+
+    def install(self, spec, prefix):
+        # Install a local version of opam for building Infer. Options
+        # taken from
+        # https://github.com/Homebrew/homebrew-core/blob/master/Formula/infer.rb
+        opamroot = join_path('.', 'opamroot')
+        mkdirp(opamroot)
+        env['OPAMROOT'] = opamroot
+        opam = Executable(join_path(self.spec['opam'].prefix.bin, 'opam'))
+        opam('init', '--no-setup')
+
+        # Detect OCaml version used in Infer build scropt
+        build_file_path = join_path('.', 'build-infer.sh')
+        f = open(build_file_path, 'r')
+        pattern = re.compile('OCAML_VERSION=\${OCAML_VERSION:-\"([^\"]+)\"}')
+        ocaml_version = pattern.search(f.read()).group(1)
+        ocaml_version_number = ocaml_version.split('+')[0]
+        f.close()
+
+        comp_file_path = join_path(opamroot, 'compilers', ocaml_version_number,
+                                   ocaml_version, ocaml_version + '.comp')
+        comp_file = FileFilter(comp_file_path)
+
+        # Add the same options Homebrew does for the configure step
+        configure_string = r'["./configure"'
+        configure_no_graph_string = r'["./configure" "-no-graph"'
+        comp_file.filter(re.escape(configure_string),
+                         configure_no_graph_string)
+
+        # Substitution to ensure `infer --version` reports a release
+        # version number; see
+        # https://github.com/Homebrew/homebrew-core/blob/master/Formula/infer.rb
+        infer_ml_in = join_path('infer', 'src', 'base', 'Version.ml.in')
+        infer_file = FileFilter(infer_ml_in)
+        is_release_string = 'let is_release = is_yes "@IS_RELEASE_TRUE@"'
+        is_release_true_string = 'let is_release = true'
+        infer_file.filter(re.escape(is_release_string),
+                          re.escape(is_release_true_string))
+
+        # TODO: Add Java support; for now, support for C, C++, and
+        # Objective-C is the focus of this package
+        target_platform = "clang"
+
+        build_infer = Executable(build_file_path)
+        build_infer(target_platform, '--yes')
+
+        opam('config', 'exec',
+             '--switch=infer-{0}'.format(ocaml_version), '--', 'make',
+             'install')
+
+        return
```

Comment:
Can you please remove the empty `return` statement?

Reply:
Done.